### PR TITLE
A compiler warning and cosmetic improvements

### PR DESCRIPTION
**Pull request by lyakh**
The compiler warning only triggers in Zephyr builds, others are all cosmetic improvements

### CODE REVIEW

File: src/audio/kpb.c [hunk 1]
```diff
@@ -1113,9 +1113,9 @@ static void kpb_init_draining(struct comp_dev *dev, struct kpb_client *cli)
 							   host_period_size / bytes_per_ms) /
 					 KPB_DRAIN_NUM_OF_PPL_PERIODS_AT_ONCE;
 			period_bytes_limit = host_period_size;
-			comp_info(dev, "kpb_init_draining(): sync_draining_mode selected with interval %d [uS].",
-				  drain_interval * 1000 /
-				  clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1));
+			comp_info(dev, "kpb_init_draining(): sync_draining_mode selected with interval %u [uS].",
+				  (unsigned int)(drain_interval * 1000 /
+						 clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1)));
 		} else {
```
Comment (iuliana-prodan) [inline src/audio/kpb.c:1119]: @lyakh I'm also having some warnings, on trace, when building SOF with Zephyr:
```
In file included from sof/sof/zephyr/../src/include/sof/lib/dai.h:27,
                 from sof/sof/src/lib/dai.c:7:
sof/sof/src/lib/dai.c: In function 'dai_group_get':
sof/sof/zephyr/include/sof/trace/trace.h:35:4: warning: implicit declaration of function 'printk'; did you mean 'cbprintf'? [-Wimplicit-function-declaration]
   35 |    printk("%llu: " format "\n", platform_timer_get(NULL), \
      |    ^~~~~~
sof/sof/zephyr/include/sof/trace/../../../../src/include/sof/trace/trace.h:287:2: note: in expansion of macro '_log_message'
  287 |  _log_message(trace_log_filtered, true, LOG_LEVEL_CRITICAL, class, ctx, id_1, \
      |  ^~~~~~~~~~~~
sof/sof/zephyr/include/sof/trace/../../../../src/include/sof/trace/trace.h:290:2: note: in expansion of macro '_trace_error_with_ids'
  290 |  _trace_error_with_ids(class, ctx, id_1, id_2, format, ##__VA_ARGS__)
      |  ^~~~~~~~~~~~~~~~~~~~~
sof/sof/zephyr/include/sof/trace/../../../../src/include/sof/trace/trace.h:374:2: note: in expansion of macro 'trace_error_with_ids'
  374 |  trace_error_with_ids(_TRACE_INV_CLASS, ctx, \
      |  ^~~~~~~~~~~~~~~~~~~~
sof/sof/src/lib/dai.c:85:3: note: in expansion of macro 'tr_err'
   85 |   tr_err(&dai_tr, "dai_group_get(): invalid group_id %u",
      |   ^~~~~~
```
Any idea how can I get rid of them?
Do I need to do some LOGGING configuration for Zephyr?
Thanks!

File: src/schedule/ll_schedule.c [hunk 2]
```diff
@@ -177,8 +176,8 @@ static void schedule_ll_client_reschedule(struct ll_schedule_data *sch)
 	/* rearm only if there is work to do */
 	if (atomic_read(&sch->domain->total_num_tasks)) {
 		/* traverse to set timer according to the earliest task */
-		list_for_item_safe(wlist, tlist, &sch->tasks) {
-			task = container_of(wlist, struct task, list);
+		list_for_item(tlist, &sch->tasks) {
```
Comment (lgirdwood) [inline src/schedule/ll_schedule.c:179]: Can anyone else modify the list here ? i.e. are IRQs OFF

File: src/audio/kpb.c (hunk 1 above)
Comment (lyakh) [reply src/audio/kpb.c:1119]: @iuliana-prodan in fact that looks like a bug in zephyr/include/sof/trace/trace.h to me. It should include sys/printk.h under `#ifdef __ZEPHYR__`, would you like to submit a fix?

File: src/schedule/ll_schedule.c (hunk 2 above)
Comment (lyakh) [reply src/schedule/ll_schedule.c:179]: `list_for_item_safe()` doesn't protect against others modifying the list, it only protects against the list being modified in the loop.

File: src/audio/kpb.c (hunk 1 above)
Comment (iuliana-prodan) [reply src/audio/kpb.c:1119]: @lyakh ok, sure, I can do it.
Let me check it and I'll submit the fix.
Thanks!

**Review by lgirdwood — APPROVED** [review]

**Comment by lyakh** [comment]
SOFCI TEST

**Review by iuliana-prodan — APPROVED** [review]

**Pull request merged**